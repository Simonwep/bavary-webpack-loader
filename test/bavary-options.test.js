import {compile, execute, getCompiler, readAsset} from './helper';

describe('Options from @bavary/core', () => {
    it('Should properly apply core-options', async () => {
        const compiler = getCompiler('complex/index.js', {
            functions: ['count'],
            locationData: {
                start: 'startsAt',
                end: 'endsAt'
            }
        });

        const stats = await compile(compiler);
        expect(execute(readAsset('main.bundle.js', compiler, stats))).toMatchSnapshot('result');
        expect(stats.compilation.warnings).toMatchSnapshot('warnings');
        expect(stats.compilation.errors).toMatchSnapshot('errors');
    });
});
