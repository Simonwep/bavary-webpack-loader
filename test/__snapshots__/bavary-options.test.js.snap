// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options from @bavary/core Should properly apply core-options: errors 1`] = `Array []`;

exports[`Options from @bavary/core Should properly apply core-options: result 1`] = `
Object {
  "decimal": "23e4",
  "num": "123",
  "numLength": 3,
  "sign": null,
}
`;

exports[`Options from @bavary/core Should properly apply core-options: warnings 1`] = `Array []`;
